drivers/meter/ade9153a/ade9153a.h

```diff
@@ -0,0 +1,1470 @@
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+#ifndef __ADE9153A_H__
+#define __ADE9153A_H__
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include <stdbool.h>
+#include <stdint.h>
+#include <string.h>
+#include "no_os_util.h"
+#include "no_os_spi.h"
+#include "no_os_gpio.h"
+#include "no_os_irq.h"
+#include "no_os_delay.h"
+#include "no_os_units.h"
+#include "no_os_alloc.h"
+#include "no_os_crc16.h"
+#include "no_os_print_log.h"
+#include <stdlib.h>
+#include <errno.h>
+#include <math.h>
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+
+/* SPI commands */
+#define ADE9153A_SPI_READ						NO_OS_BIT(3)
+
+/* Lock write key */
+#define ADE9153A_LOCK_KEY						0x3C64
+
+/* Unlock write key */
+#define ADE9153A_UNLOCK_KEY						0x4AD1
+
+/* COMPMODE val */
+#define ADE9153A_COMPMODEVAL					0x0005
+
+/* RUN */
+#define ADE9153A_RUN							0x0001
+#define ADE9153A_STOP							0x0000
+
+/* Version product */
+#define ADE9153A_VERSION						0x9153a
+
+/* Reset Delay */
+#define ADE9153A_RESET_DEL						0x1F4
+
+/* Address range of 16 bit registers */
+#define ADE9153A_START_16ADDR					0x473
+#define ADE9153A_END_16ADDR						0x600
```

Comment:
there is something wrong with these TAB indentations, if i use the View Raw file functionality on Github they still get displayed badly, so this is definitely something to fix